scripts/buildTs.mjs

```diff
@@ -84,6 +90,60 @@ packagesWithTs.forEach(({packageDir, pkg}) => {
       )}\nExpected:\n\n${jestDependenciesOfPackage.join('\n')}`,
     );
   }
+
+  let hasJestTestUtils = Object.keys(pkg.dependencies || {}).includes(
+    '@jest/test-utils',
+  );
+
+  if (hasJestTestUtils) {
+    throw new Error(
+      chalk.red(
+        `Package '${pkg.name}' declares '@jest/test-utils' as dependency, but it must be declared as dev dependency`,
+      ),
+    );
+  }
+
+  hasJestTestUtils = Object.keys(pkg.devDependencies || {}).includes(
+    '@jest/test-utils',
+  );
+
+  const tsConfigPaths = glob.sync('src/**/tsconfig.json', {
```

Comment:
For example, `@jest/core` has several `__tests__` folders. Only some of them include reference to `@jest/test-utils`.